src/dawn/engine_dawn.cc

```diff
@@ -179,8 +189,253 @@ MapResult MapBuffer(const ::dawn::Device& device, const ::dawn::Buffer& buf) {
+}
+
+::dawn::SamplerDescriptor GetDefaultSamplerDescriptor() {
+  ::dawn::SamplerDescriptor desc;
+
+  float kLodMin = 0.0;
+  float kLodMax = 1000.0;
+
+  desc.minFilter = ::dawn::FilterMode::Linear;
+  desc.magFilter = ::dawn::FilterMode::Linear;
+  desc.mipmapFilter = ::dawn::FilterMode::Linear;
+  desc.addressModeU = ::dawn::AddressMode::Repeat;
+  desc.addressModeV = ::dawn::AddressMode::Repeat;
+  desc.addressModeW = ::dawn::AddressMode::Repeat;
+  desc.lodMinClamp = kLodMin;
+  desc.lodMaxClamp = kLodMax;
+  desc.compareFunction = ::dawn::CompareFunction::Never;
+
+  return desc;
+}
+
+::dawn::BufferCopyView CreateBufferCopyView(::dawn::Buffer buffer,
+                                            uint64_t offset,
+                                            uint32_t rowPitch,
+                                            uint32_t imageHeight) {
+  ::dawn::BufferCopyView bufferCopyView;
+  bufferCopyView.buffer = buffer;
+  bufferCopyView.offset = offset;
+  bufferCopyView.rowPitch = rowPitch;
+  bufferCopyView.imageHeight = imageHeight;
+
+  return bufferCopyView;
+}
+
+::dawn::TextureCopyView CreateTextureCopyView(::dawn::Texture texture,
+                                              uint32_t level,
+                                              uint32_t slice,
+                                              ::dawn::Origin3D origin) {
+  ::dawn::TextureCopyView textureCopyView;
+  textureCopyView.texture = texture;
+  textureCopyView.level = level;
+  textureCopyView.slice = slice;
+  textureCopyView.origin = origin;
+
+  return textureCopyView;
+}
+
+::dawn::BindGroupLayout MakeBindGroupLayout(
+    const ::dawn::Device& device,
+    std::initializer_list<::dawn::BindGroupLayoutBinding> bindingsInitializer) {
+  constexpr ::dawn::ShaderStageBit kNoStages{};
+
+  std::vector<::dawn::BindGroupLayoutBinding> bindings;
+  for (const ::dawn::BindGroupLayoutBinding& binding : bindingsInitializer) {
+    if (binding.visibility != kNoStages) {
+      bindings.push_back(binding);
+    }
+  }
+
+  ::dawn::BindGroupLayoutDescriptor descriptor;
+  descriptor.bindingCount = static_cast<uint32_t>(bindings.size());
+  descriptor.bindings = bindings.data();
+  return device.CreateBindGroupLayout(&descriptor);
+}
+
+::dawn::PipelineLayout MakeBasicPipelineLayout(
+    const ::dawn::Device& device,
+    const ::dawn::BindGroupLayout* bindGroupLayout) {
+  ::dawn::PipelineLayoutDescriptor descriptor;
+  if (bindGroupLayout) {
+    descriptor.bindGroupLayoutCount = 1;
+    descriptor.bindGroupLayouts = bindGroupLayout;
+  } else {
+    descriptor.bindGroupLayoutCount = 0;
+    descriptor.bindGroupLayouts = nullptr;
+  }
+  return device.CreatePipelineLayout(&descriptor);
+}
+
+::dawn::TextureView CreateDefaultDepthStencilView(
+    const ::dawn::Device& device,
+    const ::dawn::TextureFormat depth_stencil_format) {
+  ::dawn::TextureDescriptor descriptor;
+  descriptor.dimension = ::dawn::TextureDimension::e2D;
+  descriptor.size.width = 250;
```

Comment:
Follupwup: This width and height should come from the framebuffer size in the pipeline.

Maybe add a TODO for now?

Reply:
yes you're right! I fixed them.